### PR TITLE
Fixes #1416269 Add coverage to Pontoon

### PR DESCRIPTION
**Pull request by phlax**
its really helpful to see what code is/not tested in PRs, to see what is currently not tested, and to monitor over time.

### CODE REVIEW

**Comment by mathjazz** [comment]
Let's try this! IIRC @adngdb was also in favour when we discussed this recently?

Shall we also integrate it with GitHub?

**Comment by phlax** [comment]
> Shall we also integrate it with GitHub?

I added submitting to codecov in the test run - we might want to deconfigure it a bit as it can be noisy

File: requirements.txt [hunk 1]
```diff
@@ -272,3 +272,10 @@ argparse==1.3.0 \
     --hash=sha256:b3a79a23d37b5a02faa550b92cbbbebeb4aa1d77e649c3eb39c19abf5262da04
 diff-match-patch==20121119 \
     --hash=sha256:9dba5611fbf27893347349fd51cc1911cb403682a7163373adacc565d11e2e4c
+
+# ----------------------------------------------------------------------------------
+
+coverage==4.4.2	\
+    --hash=sha256:eea9135432428d3ca7ee9be86af27cb8e56243f73764a9b6c3e0bda1394916be
+codecov==2.0.9 \
+    --hash=sha256:ad82f054837b02081f86ed1eb6c04cddc029fbc734eaf92ff73da1db3a79188b
```
Comment (adngdb) [inline requirements.txt]: I assume those have no sub-requirements, can you add them to the first section of this file, in alphabetical order?

**Comment by phlax** [comment]
> I was expecting to find something in the PR checks, but there's nothing that leads to codecov.

codecov is submitted to at the end of the travis run

> how are we going to access the coverage reports?

all pull requests will have a coverage report in the "All checks have passed" (or otherwise) section

> Does it say nothing if coverage goes up or doesn't change, and only "complain" if it goes down?

dep on how we configure it, it will fail PRs that have either low coverage in the PR, or that lower overall coverage. Otherwise it will show as passing and provide a link to view the coverage report

you can also check coverage directly here https://codecov.io/gh/mozilla/pontoon/

**Comment by phlax** [comment]
@adngdb ive separated out the test requirements into a separate file, i dont think its necessary to install this if you are not running tests and checking/submitting coverage

We might want to use `extra_requires` for this, but as we dont currently use `install_requires` i figured follow existing.

File: requirements.txt [hunk 2]
```diff
@@ -117,6 +114,28 @@ https://github.com/mathjazz/silme/archive/v0.9.3.zip#egg=silme==0.9.3 \
 # Requirements with sub-dependencies.
 # ==================================================================================
 
+requests==2.18.4 \
```
Comment (adngdb) [inline requirements.txt]: Ideally this should be lower in the requirements file, in order to follow alphabetical order.

Comment (phlax) [reply requirements.txt]: cool, ill update

**Pull request merged**